apis/Google.Cloud.Translation.V2/Google.Cloud.Translation.V2/TranslationClient.cs

```diff
@@ -0,0 +1,292 @@
+    /// </para>
+    /// </remarks>
+    public abstract class TranslationClient
+    {
+        private static readonly ScopedCredentialProvider _credentialProvider =
+            new ScopedCredentialProvider(new[] { TranslateService.Scope.CloudPlatform });
+
+        /// <summary>
+        /// The underlying translation service object used by this client.
+        /// </summary>
+        /// <remarks>
+        /// The <see cref="TranslationClient"/> class only provides convenience operations built on top of
+        /// an existing service object. Any more complex operations which are not supported by this wrapper may wish
+        /// to use the same service object as the wrapper, in order to take advantage of its configuration (for authentication,
+        /// application naming etc).
+        /// </remarks>
+        public virtual TranslateService Service { get { throw new NotImplementedException(); } }
+
+        /// <summary>
+        /// The default translation model used by this client.
+        /// </summary>
+        /// <remarks>
+        /// Each <c>Translate</c> method has an optional parameter to override this.
+        /// </remarks>
+        public virtual TranslationModel DefaultModel { get { throw new NotImplementedException(); } }
+
+        /// <summary>
+        /// Translates a single item of text synchronously.
+        /// </summary>
+        /// <param name="text">The text to translate.</param>
+        /// <param name="targetLanguage">The code for the language to translate into. Must not be null.</param>
+        /// <param name="sourceLanguage">The code for the language to translate from. May be null, in which case the server
+        /// will detect the source language.</param>
+        /// <param name="model">The model to request for translation. May be null, in which case <see cref="DefaultModel"/> is
+        /// used.</param>
+        /// <returns>The translation of <paramref name="text"/>.</returns>
+        public virtual TranslationResult TranslateText(string text, string targetLanguage, string sourceLanguage = null, TranslationModel? model = null)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Translates a single item of HTML synchronously.
+        /// </summary>
+        /// <param name="html">The HTML to translate.</param>
+        /// <param name="targetLanguage">The code for the language to translate into. Must not be null.</param>
+        /// <param name="sourceLanguage">The code for the language to translate from. May be null, in which case the server
+        /// will detect the source language.</param>
+        /// <param name="model">The model to request for translation. May be null, in which case <see cref="DefaultModel"/> is
+        /// used.</param>
+        /// <returns>The translation of <paramref name="html"/>.</returns>
```

Comment:
It might be good to have remarks saying what can be translated, such as fragments, whole documents, other restrictions. Or perhaps just include a link to https://cloud.google.com/translate/markup.